specification/ai/OpenAI.Assistants/files/routes.tsp

```diff
@@ -53,8 +53,10 @@ op uploadFile(
    *            any specific data source like a file. It can be inferred in some circumstances (as when using direct
    *            file input, like curl does) but should remain configurable when using a stream or other data source
    *            lacking an a priori name.
+   *            If this value is not sent to the OpenAI backend the request will result in an error, so `filename` is
+   *            functinonally required.
    */
-  filename?: string,
+  filename: string,
```

Comment:
Does any SDK other than .NET need this "filename"?

Both Java and Python would have problem if user do send this "filename" as multipart/form-data part.
https://github.com/Azure/azure-rest-api-specs/issues/28905

Reply:
Agree with @weidongxu-microsoft, Please don't add the `filename` which is useless even harmful for Python SDK and Java SDK generation.

Reply:
yes. if the `filename` is not sent to network, and it is just a meta-data of a file, it should not be an model property. It will harm the .NET SDK generation also. we may need to remove `filename`

Reply:
Thank you for the feedback. I have reverted this change so we can move with Assistants V1 streaming and will punt the resolution of this discrepancy with the public spec in different PR.

Reply:
I find myself in agreement with you, `filename` is not part of the [public OpenAI spec](https://github.com/openai/openai-openapi/blob/893ba52242dbd5387a97b96444ee1c742cfce9bd/openapi.yaml#L8588-L8606), but the service does need a default value provided for the multipart header, AFAIK.